clang/lib/Sema/SemaBounds.cpp

```diff
@@ -655,6 +703,31 @@ namespace {
   // expressions that produce the same value as e.
   using ExprEqualMapTy = llvm::DenseMap<Expr *, EqualExprTy>;
 
+  // Describes the position of a free variable (FR).
+  enum class FVPos {
+    Lower = 0x1,    // The FR appears in (any) lower bounds.
+    Upper = 0x2,    // The FR appears in (any) upper bounds.
+    Base = Lower | Upper, // The FR appears in (any) base bounds.
```

Comment:
Instead of defining each combination here, I suggest only defining `Lower`, `Upper`, `Base`, and `Declared`, and creating the other combinations where they are needed.